src/megaapi_impl.cpp

```diff
@@ -20051,6 +20119,23 @@ void MegaApiImpl::sendPendingRequests()
             client->acknowledgeuseralerts();
             break;
         }
+        case MegaRequest::TYPE_PUBLIC_LINK_INFORMATION:
+        {
+            const char *link = request->getLink();
+            if (!link)
+            {
+                e = API_EARGS;
+                break;
+            }
+
+            e = client->folderaccess(link);
```

Comment:
This is a `NO NO NO`, @jgandres . If you check the code in that function, it will set the master key to the key of the folder link, and also the rootnode to the public handle, plus the authentication token to the public handle for API requests.